<a href="https://colab.research.google.com/github/MichaelKru92/Projekt-ML-Modelierung/blob/main/Schritt_3_Kurze_Datenexploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Ein erster Blick auf die Daten**

Der folgenden Funktion wird eine Liste mit den Dateinamen der entpackten zip-Datei übergeben.

Es erfolgt eine erste kurze Exploration der Daten:

*   Darstellung der ersten 10 Zeilen jeder Tabelle
*   Ausgabe der Datentypen, Spaltennamen und Nullwerte

In [10]:
import pandas as pd
import numpy as np
import os


CSV_PATH = os.path.join(ROHDATEN_LOCAL_PATH, "Rohdaten(CSV)")

CSV_TABLES = ["careplans.csv", "conditions.csv", "devices.csv", "disease.csv", "encounters.csv", "imaging_studies.csv", "immunizations.csv",
              "medications.csv", "observations.csv", "organizations.csv", "patients.csv", "payer_transitions.csv", "payers.csv", "procedures.csv",
              "providers.csv", "supplies.csv"]

def load_csv_to_df_overview(csv_tables=CSV_TABLES, csv_path=CSV_PATH):
    for i in csv_tables:
        csv_table_path = os.path.join(csv_path, i)
        print(f"\n\nDie ersten 10 Zeilen von Tabelle {i}:")
        display(pd.read_csv(csv_table_path).head())
        print(f"\n\nMetadaten zu Tabelle {i}:\n")
        pd.read_csv(csv_table_path).info()
        print(f"\n\nNullwerte in Tabelle {i}:\n", pd.read_csv(csv_table_path).isnull().sum())


load_csv_to_df_overview()

Downloaded to: Rohdaten/Rohdaten.zip
Extracted to: Rohdaten/Rohdaten(CSV)


Die ersten 10 Zeilen von Tabelle careplans.csv:


Id       START        STOP  \
0  b066696b-4a2f-ec29-b018-bcf7464d020c  1943-02-15         NaN   
1  8a34101d-6503-4181-6d2d-d316e5f71de0  2010-01-20         NaN   
2  b70e996d-ad15-b34b-d528-e8150d56379d  2000-01-07         NaN   
3  a91f64e4-037e-fff5-6ed0-28c3c0d04ec2  2000-01-07         NaN   
4  cfabbc6f-5086-7399-caa5-1da31e629661  2019-01-30  2019-05-27   

                                PATIENT                             ENCOUNTER  \
0  3575b903-dbd0-1d55-6146-9e8aa4ed52a5  fd59905a-4ec8-62a6-d45d-d2a458d60111   
1  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  be256df1-75cb-cbf7-0109-e0e1ae2c6348   
2  9bbbcada-7a45-92f0-6ae6-d197bcefc0d4  1b85d074-6a01-7607-27b2-f1e9f6bf305d   
3  9bbbcada-7a45-92f0-6ae6-d197bcefc0d4  a8ea7228-41bd-66d9-12f6-912d0e2dfbe4   
4  3575b903-dbd0-1d55-6146-9e8aa4ed52a5  cd0f83e5-1e70-75ec-1be9-b254e2589e49   

        CODE                                 DESCRIPTION  REASONCODE  \
0  384758001         Self-care interventions (procedure)         NaN   
1  443402002  Lifestyle education regarding hypertension  59621000.0   
2  698360004               Diabetes self management plan  44054006.0   
3  443402002  Lifestyle education regarding hypertension  59621000.0   
4   53950000                         Respiratory therapy  10509002.0   

             REASONDESCRIPTION  
0                          NaN  
1                 Hypertension  
2                     Diabetes  
3                 Hypertension  
4  Acute bronchitis (disorder)



Metadaten zu Tabelle careplans.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Id                 545 non-null    object 
 1   START              545 non-null    object 
 2   STOP               381 non-null    object 
 3   PATIENT            545 non-null    object 
 4   ENCOUNTER          545 non-null    object 
 5   CODE               545 non-null    int64  
 6   DESCRIPTION        545 non-null    object 
 7   REASONCODE         522 non-null    float64
 8   REASONDESCRIPTION  522 non-null    object 
dtypes: float64(1), int64(1), object(7)
memory usage: 38.4+ KB


Nullwerte in Tabelle careplans.csv:
 Id                     0
START                  0
STOP                 164
PATIENT                0
ENCOUNTER              0
CODE                   0
DESCRIPTION            0
REASONCODE            23
REASONDESCRIPTION     23
dt

START STOP                               PATIENT  \
0  1946-01-11  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
1  1955-01-17  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
2  1999-06-06  NaN  17f0c6d9-8931-8839-66cb-3ca6fb066d3e   
3  1999-11-25  NaN  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5   
4  1999-11-25  NaN  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5   

                              ENCOUNTER       CODE  \
0  6e0279b7-c477-a2c6-e75c-a911f03a264c  232353008   
1  40ca71e3-5668-dd3a-d7e1-209a43835384  162864005   
2  dbaad3dc-c5a8-9daf-5d5f-330a8022263f  162864005   
3  459c64dc-6783-3e8c-91d1-177c47c7258f  128613002   
4  459c64dc-6783-3e8c-91d1-177c47c7258f  703151001   

                                         DESCRIPTION  
0  Perennial allergic rhinitis with seasonal vari...  
1            Body mass index 30+ - obesity (finding)  
2            Body mass index 30+ - obesity (finding)  
3                                   Seizure disorder  
4              History of single seizure (situation)



Metadaten zu Tabelle conditions.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1601 entries, 0 to 1600
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   START        1601 non-null   object
 1   STOP         1097 non-null   object
 2   PATIENT      1601 non-null   object
 3   ENCOUNTER    1601 non-null   object
 4   CODE         1601 non-null   int64 
 5   DESCRIPTION  1601 non-null   object
dtypes: int64(1), object(5)
memory usage: 75.2+ KB


Nullwerte in Tabelle conditions.csv:
 START            0
STOP           504
PATIENT          0
ENCOUNTER        0
CODE             0
DESCRIPTION      0
dtype: int64


Die ersten 10 Zeilen von Tabelle devices.csv:


START                  STOP  \
0  2017-02-01T17:55:15Z                   NaN   
1  2020-03-17T16:11:54Z  2020-03-18T17:19:54Z   
2  2020-03-17T16:11:54Z  2020-03-17T16:31:54Z   
3  2020-03-12T22:31:39Z  2020-03-16T03:23:39Z   
4  2020-03-12T22:31:39Z  2020-03-12T22:56:39Z   

                                PATIENT                             ENCOUNTER  \
0  7a12d026-85c5-6543-0667-c04bb787dd22  ce5f125f-6038-6288-0cc9-0888ab46fb95   
1  560a4b1a-5d16-3a8d-144a-e5f36f801a4e  14a6a742-8e42-e8eb-48b4-20be5c6b8b64   
2  560a4b1a-5d16-3a8d-144a-e5f36f801a4e  14a6a742-8e42-e8eb-48b4-20be5c6b8b64   
3  a7910f38-b9fa-0e0b-dfd7-1299f957a88e  65921960-8d59-43a7-b520-421eee5a6071   
4  a7910f38-b9fa-0e0b-dfd7-1299f957a88e  65921960-8d59-43a7-b520-421eee5a6071   

        CODE                                      DESCRIPTION  \
0  706004007  Implantable cardiac pacemaker (physical object)   
1  449071006          Mechanical ventilator (physical object)   
2  448907002              Videolaryngoscope (physical object)   
3  449071006          Mechanical ventilator (physical object)   
4  448907002              Videolaryngoscope (physical object)   

                                                 UDI  
0  (01)91395072195002(11)170111(17)420126(10)8396...  
1  (01)82939641752982(11)200225(17)450311(10)7220...  
2  (01)70421022136053(11)200225(17)450311(10)1736...  
3  (01)67781150387183(11)200220(17)450306(10)6496...  
4  (01)68373751803457(11)200220(17)450306(10)1751...



Metadaten zu Tabelle devices.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26 entries, 0 to 25
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   START        26 non-null     object
 1   STOP         15 non-null     object
 2   PATIENT      26 non-null     object
 3   ENCOUNTER    26 non-null     object
 4   CODE         26 non-null     int64 
 5   DESCRIPTION  26 non-null     object
 6   UDI          26 non-null     object
dtypes: int64(1), object(6)
memory usage: 1.6+ KB


Nullwerte in Tabelle devices.csv:
 START           0
STOP           11
PATIENT         0
ENCOUNTER       0
CODE            0
DESCRIPTION     0
UDI             0
dtype: int64


Die ersten 10 Zeilen von Tabelle disease.csv:


START STOP                               PATIENT  \
0  1943-02-26  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
1  1943-02-26  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
2  1943-02-26  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
3  1943-02-26  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
4  1943-02-26  NaN  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   

                              ENCOUNTER       CODE              DESCRIPTION  
0  de8159bf-90ae-c9a6-dd8d-f1de7b7972a7  419474003         Allergy to mould  
1  de8159bf-90ae-c9a6-dd8d-f1de7b7972a7  232350006  House dust mite allergy  
2  de8159bf-90ae-c9a6-dd8d-f1de7b7972a7  232347008  Dander (animal) allergy  
3  de8159bf-90ae-c9a6-dd8d-f1de7b7972a7  418689008  Allergy to grass pollen  
4  de8159bf-90ae-c9a6-dd8d-f1de7b7972a7  419263009   Allergy to tree pollen



Metadaten zu Tabelle disease.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   START        51 non-null     object
 1   STOP         3 non-null      object
 2   PATIENT      51 non-null     object
 3   ENCOUNTER    51 non-null     object
 4   CODE         51 non-null     int64 
 5   DESCRIPTION  51 non-null     object
dtypes: int64(1), object(5)
memory usage: 2.5+ KB


Nullwerte in Tabelle disease.csv:
 START           0
STOP           48
PATIENT         0
ENCOUNTER       0
CODE            0
DESCRIPTION     0
dtype: int64


Die ersten 10 Zeilen von Tabelle encounters.csv:


Id                 START  \
0  fd59905a-4ec8-62a6-d45d-d2a458d60111  1943-02-15T11:40:19Z   
1  de8159bf-90ae-c9a6-dd8d-f1de7b7972a7  1943-02-26T11:40:19Z   
2  6e0279b7-c477-a2c6-e75c-a911f03a264c  1946-01-11T11:40:19Z   
3  40ca71e3-5668-dd3a-d7e1-209a43835384  1955-01-17T11:40:19Z   
4  905865c3-d0fd-aa37-b2dd-8f297e43a089  2011-04-11T11:40:19Z   

                   STOP                               PATIENT  \
0  1943-02-15T11:55:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
1  1943-02-26T11:55:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
2  1946-01-11T11:55:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
3  1955-01-17T11:55:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
4  2011-04-11T11:55:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   

                           ORGANIZATION                              PROVIDER  \
0  49318f80-bd8b-3fc7-a096-ac43088b0c12  4979a06f-2a09-3130-be6b-4cf2c82a2641   
1  49318f80-bd8b-3fc7-a096-ac43088b0c12  4979a06f-2a09-3130-be6b-4cf2c82a2641   
2  49318f80-bd8b-3fc7-a096-ac43088b0c12  4979a06f-2a09-3130-be6b-4cf2c82a2641   
3  3e3b089c-4e4f-3831-8607-cbf21c17f65e  17d93682-42c5-3341-970a-1b1dee5c4224   
4  3e3b089c-4e4f-3831-8607-cbf21c17f65e  17d93682-42c5-3341-970a-1b1dee5c4224   

                                  PAYER ENCOUNTERCLASS       CODE  \
0  5059a55e-5d6e-34d1-b6cb-d83d16e57bcf     ambulatory  185347001   
1  5059a55e-5d6e-34d1-b6cb-d83d16e57bcf     ambulatory  185347001   
2  5059a55e-5d6e-34d1-b6cb-d83d16e57bcf     ambulatory  185345009   
3  5059a55e-5d6e-34d1-b6cb-d83d16e57bcf       wellness  410620009   
4  7caa7254-5050-3b5e-9eae-bd5ea30e809c       wellness  162673000   

                                  DESCRIPTION  BASE_ENCOUNTER_COST  \
0                       Encounter for problem                77.49   
1                       Encounter for problem                77.49   
2                       Encounter for symptom                77.49   
3                Well child visit (procedure)               129.16   
4  General examination of patient (procedure)               129.16   

   TOTAL_CLAIM_COST  PAYER_COVERAGE   REASONCODE  \
0             77.49            7.49          NaN   
1             77.49            7.49          NaN   
2             77.49            7.49  232353008.0   
3            129.16           59.16          NaN   
4            129.16          129.16          NaN   

                                   REASONDESCRIPTION  
0                                                NaN  
1                                                NaN  
2  Perennial allergic rhinitis with seasonal vari...  
3                                                NaN  
4                                                NaN



Metadaten zu Tabelle encounters.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7768 entries, 0 to 7767
Data columns (total 15 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Id                   7768 non-null   object 
 1   START                7768 non-null   object 
 2   STOP                 7768 non-null   object 
 3   PATIENT              7768 non-null   object 
 4   ORGANIZATION         7768 non-null   object 
 5   PROVIDER             7768 non-null   object 
 6   PAYER                7768 non-null   object 
 7   ENCOUNTERCLASS       7768 non-null   object 
 8   CODE                 7768 non-null   int64  
 9   DESCRIPTION          7768 non-null   object 
 10  BASE_ENCOUNTER_COST  7768 non-null   float64
 11  TOTAL_CLAIM_COST     7768 non-null   float64
 12  PAYER_COVERAGE       7768 non-null   float64
 13  REASONCODE           1827 non-null   float64
 14  REASONDESCRIPTION    1827 non-null   object 
dty

Id                  DATE  \
0  91de3567-3a8f-a1e8-0d57-7ae3dbd71364  2017-05-04T16:57:44Z   
1  57a92f9f-fe76-a052-724d-ebe5706c78e0  2013-05-08T11:04:40Z   
2  792c8ed0-93e3-3f30-cdbb-6548d6a256eb  2017-07-02T04:12:24Z   
3  3832cbeb-9159-4a48-6db6-8f58555435cb  2019-02-22T03:00:33Z   
4  f152aab1-ab35-c631-0f29-15417abf6405  2013-11-29T02:00:18Z   

                                PATIENT                             ENCOUNTER  \
0  27b0d72c-f2fb-7e25-38c0-7d5120ebbedf  613d3d12-7b27-9b05-c8a3-dc53d725d4de   
1  d6d70096-ab54-ad6d-ec5a-0a5d2639556a  8c23eae9-d5c3-152e-eb04-8ef541a1e04e   
2  70b43d3b-b0b5-1ffa-8e48-7a6c011cb5cb  dd9175d7-af13-8955-8220-a0a3c9235817   
3  8ba5469f-0e7d-8f3a-2c26-e5996ffcaa90  a6a74ded-0ed7-379e-aa21-f4e22acfbdd0   
4  6f74fd66-9dbd-b126-0c0f-55b798c2a014  13a26b78-7cab-321b-10f2-0bb6053641a1   

   BODYSITE_CODE BODYSITE_DESCRIPTION MODALITY_CODE MODALITY_DESCRIPTION  \
0       51299004             Clavicle            DX  Digital Radiography   
1       51299004             Clavicle            DX  Digital Radiography   
2       51299004             Clavicle            DX  Digital Radiography   
3        8205005                Wrist            DX  Digital Radiography   
4         344001                Ankle            DX  Digital Radiography   

                      SOP_CODE              SOP_DESCRIPTION  
0  1.2.840.10008.5.1.4.1.1.1.1  Digital X-Ray Image Storage  
1  1.2.840.10008.5.1.4.1.1.1.1  Digital X-Ray Image Storage  
2  1.2.840.10008.5.1.4.1.1.1.1  Digital X-Ray Image Storage  
3  1.2.840.10008.5.1.4.1.1.1.1  Digital X-Ray Image Storage  
4  1.2.840.10008.5.1.4.1.1.1.1  Digital X-Ray Image Storage



Metadaten zu Tabelle imaging_studies.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 75 entries, 0 to 74
Data columns (total 10 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Id                    75 non-null     object
 1   DATE                  75 non-null     object
 2   PATIENT               75 non-null     object
 3   ENCOUNTER             75 non-null     object
 4   BODYSITE_CODE         75 non-null     int64 
 5   BODYSITE_DESCRIPTION  75 non-null     object
 6   MODALITY_CODE         75 non-null     object
 7   MODALITY_DESCRIPTION  75 non-null     object
 8   SOP_CODE              75 non-null     object
 9   SOP_DESCRIPTION       75 non-null     object
dtypes: int64(1), object(9)
memory usage: 6.0+ KB


Nullwerte in Tabelle imaging_studies.csv:
 Id                      0
DATE                    0
PATIENT                 0
ENCOUNTER               0
BODYSITE_CODE           0
BODYSITE_DESCRIPTION    0
M

DATE                               PATIENT  \
0  2011-04-11T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
1  2012-04-16T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
2  2013-04-22T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
3  2013-04-22T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
4  2014-04-28T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   

                              ENCOUNTER  CODE  \
0  905865c3-d0fd-aa37-b2dd-8f297e43a089   140   
1  a30d46d7-823d-f721-60d5-58c3f9b4e519   140   
2  d37cc838-ee0b-2e4d-8c96-db68d01e0b29   140   
3  d37cc838-ee0b-2e4d-8c96-db68d01e0b29   113   
4  e7defffa-fc95-542e-03a3-485e20469db4   140   

                                         DESCRIPTION  BASE_COST  
0  Influenza  seasonal  injectable  preservative ...     140.52  
1  Influenza  seasonal  injectable  preservative ...     140.52  
2  Influenza  seasonal  injectable  preservative ...     140.52  
3                       Td (adult) preservative free     140.52  
4  Influenza  seasonal  injectable  preservative ...     140.52



Metadaten zu Tabelle immunizations.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1671 entries, 0 to 1670
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   DATE         1671 non-null   object 
 1   PATIENT      1671 non-null   object 
 2   ENCOUNTER    1671 non-null   object 
 3   CODE         1671 non-null   int64  
 4   DESCRIPTION  1671 non-null   object 
 5   BASE_COST    1671 non-null   float64
dtypes: float64(1), int64(1), object(4)
memory usage: 78.5+ KB


Nullwerte in Tabelle immunizations.csv:
 DATE           0
PATIENT        0
ENCOUNTER      0
CODE           0
DESCRIPTION    0
BASE_COST      0
dtype: int64


Die ersten 10 Zeilen von Tabelle medications.csv:


START                  STOP  \
0  1946-01-11T11:40:19Z                   NaN   
1  2010-01-20T00:48:18Z  2010-10-20T00:48:18Z   
2  2010-02-19T00:48:18Z  2010-10-20T00:48:18Z   
3  2010-04-20T00:48:18Z  2010-10-20T00:48:18Z   
4  2010-10-20T00:48:18Z  2010-12-15T00:48:18Z   

                                PATIENT                                 PAYER  \
0  3575b903-dbd0-1d55-6146-9e8aa4ed52a5  5059a55e-5d6e-34d1-b6cb-d83d16e57bcf   
1  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  047f6ec3-6215-35eb-9608-f9dda363a44c   
2  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  047f6ec3-6215-35eb-9608-f9dda363a44c   
3  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  047f6ec3-6215-35eb-9608-f9dda363a44c   
4  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  b1c428d6-4f07-31e0-90f0-68ffa6ff8c76   

                              ENCOUNTER     CODE  \
0  6e0279b7-c477-a2c6-e75c-a911f03a264c  1049630   
1  be256df1-75cb-cbf7-0109-e0e1ae2c6348   314076   
2  78f3c03e-8264-77b9-dc70-16fd00193b8d   310798   
3  6f149158-46da-7ad4-040d-fd9786e66508   308136   
4  583a6d5e-4bda-34f7-06a5-97051581836f   310798   

                                       DESCRIPTION  BASE_COST  PAYER_COVERAGE  \
0  diphenhydrAMINE Hydrochloride 25 MG Oral Tablet       8.80             0.0   
1                     lisinopril 10 MG Oral Tablet       0.02             0.0   
2            Hydrochlorothiazide 25 MG Oral Tablet       0.01             0.0   
3                    amLODIPine 2.5 MG Oral Tablet       0.02             0.0   
4            Hydrochlorothiazide 25 MG Oral Tablet       0.01             0.0   

   DISPENSES  TOTALCOST  REASONCODE REASONDESCRIPTION  
0        911    8016.80         NaN               NaN  
1        273       5.46  59621000.0      Hypertension  
2          8       0.08  59621000.0      Hypertension  
3        183       3.66  59621000.0      Hypertension  
4          1       0.01  59621000.0      Hypertension



Metadaten zu Tabelle medications.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8536 entries, 0 to 8535
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   START              8536 non-null   object 
 1   STOP               8232 non-null   object 
 2   PATIENT            8536 non-null   object 
 3   PAYER              8536 non-null   object 
 4   ENCOUNTER          8536 non-null   object 
 5   CODE               8536 non-null   int64  
 6   DESCRIPTION        8536 non-null   object 
 7   BASE_COST          8536 non-null   float64
 8   PAYER_COVERAGE     8536 non-null   float64
 9   DISPENSES          8536 non-null   int64  
 10  TOTALCOST          8536 non-null   float64
 11  REASONCODE         6829 non-null   float64
 12  REASONDESCRIPTION  6829 non-null   object 
dtypes: float64(4), int64(2), object(7)
memory usage: 867.1+ KB


Nullwerte in Tabelle medications.csv:
 START                   0
S

DATE                               PATIENT  \
0  2011-04-11T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
1  2011-04-11T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
2  2011-04-11T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
3  2011-04-11T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
4  2011-04-11T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   

                              ENCOUNTER     CODE  \
0  905865c3-d0fd-aa37-b2dd-8f297e43a089   8302-2   
1  905865c3-d0fd-aa37-b2dd-8f297e43a089  72514-3   
2  905865c3-d0fd-aa37-b2dd-8f297e43a089  29463-7   
3  905865c3-d0fd-aa37-b2dd-8f297e43a089  39156-5   
4  905865c3-d0fd-aa37-b2dd-8f297e43a089   8462-4   

                                         DESCRIPTION  VALUE    UNITS     TYPE  
0                                        Body Height  152.6       cm  numeric  
1  Pain severity - 0-10 verbal numeric rating [Sc...    2.0  {score}  numeric  
2                                        Body Weight   65.9       kg  numeric  
3                                    Body Mass Index   28.3    kg/m2  numeric  
4                           Diastolic Blood Pressure   82.0   mm[Hg]  numeric



Metadaten zu Tabelle observations.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47331 entries, 0 to 47330
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   DATE         47331 non-null  object
 1   PATIENT      47331 non-null  object
 2   ENCOUNTER    44148 non-null  object
 3   CODE         47331 non-null  object
 4   DESCRIPTION  47331 non-null  object
 5   VALUE        47331 non-null  object
 6   UNITS        45174 non-null  object
 7   TYPE         47331 non-null  object
dtypes: object(8)
memory usage: 2.9+ MB


Nullwerte in Tabelle observations.csv:
 DATE              0
PATIENT           0
ENCOUNTER      3183
CODE              0
DESCRIPTION       0
VALUE             0
UNITS          2157
TYPE              0
dtype: int64


Die ersten 10 Zeilen von Tabelle organizations.csv:


Id                           NAME  \
0  ef58ea08-d883-3957-8300-150554edc8fb  HEALTHALLIANCE HOSPITALS  INC   
1  69176529-fd1f-3b3f-abce-a0a3626769eb          MOUNT AUBURN HOSPITAL   
2  5e765f2b-e908-3888-9fc7-df2cb87beb58       STURDY MEMORIAL HOSPITAL   
3  f1fbcbfb-fcfa-3bd2-b7f4-df20f1b3c3a4      LAWRENCE GENERAL HOSPITAL   
4  e002090d-4e92-300e-b41e-7d1f21dee4c6      CAMBRIDGE HEALTH ALLIANCE   

                   ADDRESS        CITY STATE    ZIP        LAT        LON  \
0         60 HOSPITAL ROAD  LEOMINSTER    MA  01453  42.520838 -71.770876   
1  330 MOUNT AUBURN STREET   CAMBRIDGE    MA  02138  42.375967 -71.118275   
2          211 PARK STREET   ATTLEBORO    MA  02703  41.931653 -71.294503   
3       ONE GENERAL STREET    LAWRENCE    MA  01842  42.700273 -71.161357   
4    1493 CAMBRIDGE STREET   CAMBRIDGE    MA  02138  42.375967 -71.118275   

        PHONE   REVENUE  UTILIZATION  
0  9784662000  12528.37          107  
1  6174923500  13018.89          131  
2  5082225200    154.98            2  
3  9786834000  63209.17          607  
4  6176652300  15937.96          149



Metadaten zu Tabelle organizations.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 216 entries, 0 to 215
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Id           216 non-null    object 
 1   NAME         216 non-null    object 
 2   ADDRESS      216 non-null    object 
 3   CITY         216 non-null    object 
 4   STATE        216 non-null    object 
 5   ZIP          216 non-null    object 
 6   LAT          216 non-null    float64
 7   LON          216 non-null    float64
 8   PHONE        191 non-null    object 
 9   REVENUE      216 non-null    float64
 10  UTILIZATION  216 non-null    int64  
dtypes: float64(3), int64(1), object(7)
memory usage: 18.7+ KB


Nullwerte in Tabelle organizations.csv:
 Id              0
NAME            0
ADDRESS         0
CITY            0
STATE           0
ZIP             0
LAT             0
LON             0
PHONE          25
REVENUE         0
UTILIZATION     0
dt

Id   BIRTHDATE DEATHDATE          SSN  \
0  27b0d72c-f2fb-7e25-38c0-7d5120ebbedf  1945-04-17       NaN  999-93-9618   
1  17f0c6d9-8931-8839-66cb-3ca6fb066d3e  1959-05-31       NaN  999-17-6461   
2  aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  1991-11-27       NaN  999-29-1688   
3  e92765c7-1c4b-9ee7-dbc8-5300fcb40a54  2001-11-18       NaN  999-30-3448   
4  3575b903-dbd0-1d55-6146-9e8aa4ed52a5  1941-12-22       NaN  999-73-1683   

     DRIVERS    PASSPORT PREFIX      FIRST            LAST SUFFIX  ...  \
0  S99971149   X1878420X   Mrs.   Emmie273         Schoen8    NaN  ...   
1  S99999085  X20851735X    Mr.    Karl184      Stroman228    NaN  ...   
2  S99979872  X48771954X    Mr.   Monty345      Krajcik437    NaN  ...   
3  S99973612         NaN    Mr.    Hugo693     Valencia279    NaN  ...   
4  S99984672  X13615296X   Mrs.  Hayley136  Schamberger479    NaN  ...   

                      BIRTHPLACE                    ADDRESS         CITY  \
0   Lexington  Massachusetts  US          1056 Hegmann Mall     Kingston   
1  Sturbridge  Massachusetts  US  1080 Boyer Brook Suite 22       Sutton   
2   Watertown  Massachusetts  US      657 Fay Road Suite 26    Haverhill   
3  Caracas  Capital District  VE    762 Cormier Forge Apt 2       Boston   
4     Everett  Massachusetts  US  217 Nitzsche Green Apt 80  Easthampton   

           STATE            COUNTY     ZIP        LAT        LON  \
0  Massachusetts   Plymouth County  2364.0  41.996106 -70.786205   
1  Massachusetts  Worcester County     NaN  42.126130 -71.749848   
2  Massachusetts      Essex County  1830.0  42.763237 -71.117704   
3  Massachusetts    Suffolk County  2108.0  42.391083 -71.041182   
4  Massachusetts  Hampshire County  1027.0  42.282323 -72.656783   

  HEALTHCARE_EXPENSES HEALTHCARE_COVERAGE  
0            63141.11             3176.78  
1          1333971.83             4339.05  
2           735728.62             4015.40  
3            58484.07             2829.75  
4          1546947.37             9558.05  

[5 rows x 25 columns]



Metadaten zu Tabelle patients.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 135 entries, 0 to 134
Data columns (total 25 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Id                   135 non-null    object 
 1   BIRTHDATE            135 non-null    object 
 2   DEATHDATE            35 non-null     object 
 3   SSN                  135 non-null    object 
 4   DRIVERS              119 non-null    object 
 5   PASSPORT             110 non-null    object 
 6   PREFIX               113 non-null    object 
 7   FIRST                135 non-null    object 
 8   LAST                 135 non-null    object 
 9   SUFFIX               1 non-null      object 
 10  MAIDEN               32 non-null     object 
 11  MARITAL              99 non-null     object 
 12  RACE                 135 non-null    object 
 13  ETHNICITY            135 non-null    object 
 14  GENDER               135 non-null    object 
 15  BI

PATIENT  START_YEAR  END_YEAR  \
0  3575b903-dbd0-1d55-6146-9e8aa4ed52a5        1941      1958   
1  3575b903-dbd0-1d55-6146-9e8aa4ed52a5        1959      1959   
2  3575b903-dbd0-1d55-6146-9e8aa4ed52a5        1960      1964   
3  3575b903-dbd0-1d55-6146-9e8aa4ed52a5        1965      1965   
4  3575b903-dbd0-1d55-6146-9e8aa4ed52a5        1966      1966   

                                  PAYER OWNERSHIP  
0  7c4411ce-02f1-39b5-b9ec-dfbea9ad3c1a  Guardian  
1  047f6ec3-6215-35eb-9608-f9dda363a44c      Self  
2  7c4411ce-02f1-39b5-b9ec-dfbea9ad3c1a      Self  
3  5059a55e-5d6e-34d1-b6cb-d83d16e57bcf      Self  
4  7c4411ce-02f1-39b5-b9ec-dfbea9ad3c1a      Self



Metadaten zu Tabelle payer_transitions.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   PATIENT     395 non-null    object
 1   START_YEAR  395 non-null    int64 
 2   END_YEAR    395 non-null    int64 
 3   PAYER       395 non-null    object
 4   OWNERSHIP   351 non-null    object
dtypes: int64(2), object(3)
memory usage: 15.6+ KB


Nullwerte in Tabelle payer_transitions.csv:
 PATIENT        0
START_YEAR     0
END_YEAR       0
PAYER          0
OWNERSHIP     44
dtype: int64


Die ersten 10 Zeilen von Tabelle payers.csv:


Id                    NAME  \
0  b3221cfc-24fb-339e-823d-bc4136cbc4ed           Dual Eligible   
1  7caa7254-5050-3b5e-9eae-bd5ea30e809c                Medicare   
2  7c4411ce-02f1-39b5-b9ec-dfbea9ad3c1a                Medicaid   
3  d47b3510-2895-3b70-9897-342d681c769d                  Humana   
4  6e2f1a2d-27bd-3701-8d08-dae202c58632  Blue Cross Blue Shield   

              ADDRESS        CITY STATE_HEADQUARTERED      ZIP  \
0  7500 Security Blvd   Baltimore                  MD  21244.0   
1  7500 Security Blvd   Baltimore                  MD  21244.0   
2  7500 Security Blvd   Baltimore                  MD  21244.0   
3    500 West Main St  Louisville                  KY  40018.0   
4      Michigan Plaza     Chicago                  IL  60007.0   

            PHONE  AMOUNT_COVERED  AMOUNT_UNCOVERED     REVENUE  ...  \
0  1-877-267-2323        16340.20          18833.47    413000.0  ...   
1  1-800-633-4227      1070850.80         295557.17   1512250.0  ...   
2  1-877-267-2323       237538.03         181925.58   2406250.0  ...   
3  1-844-330-7799        58622.92          83709.75  13940000.0  ...   
4  1-800-262-2583       299454.41         237276.62  16763400.0  ...   

   UNCOVERED_ENCOUNTERS  COVERED_MEDICATIONS  UNCOVERED_MEDICATIONS  \
0                     0                  128                      0   
1                     0                 4767                      0   
2                     0                  472                      0   
3                   123                    0                    342   
4                     0                 1648                      0   

   COVERED_PROCEDURES  UNCOVERED_PROCEDURES  COVERED_IMMUNIZATIONS  \
0                 135                     0                     84   
1                2693                     0                    578   
2                5983                     0                    778   
3                1015                    59                    798   
4                1323                     0                    664   

   UNCOVERED_IMMUNIZATIONS  UNIQUE_CUSTOMERS  QOLS_AVG  MEMBER_MONTHS  
0                        0                 6  0.745076            852  
1                        0                37  0.849558           6252  
2                        0                45  0.878953           9648  
3                       18                31  0.960398           8268  
4                        0                30  0.885813           9396  

[5 rows x 21 columns]



Metadaten zu Tabelle payers.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 21 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Id                       10 non-null     object 
 1   NAME                     10 non-null     object 
 2   ADDRESS                  9 non-null      object 
 3   CITY                     9 non-null      object 
 4   STATE_HEADQUARTERED      9 non-null      object 
 5   ZIP                      9 non-null      float64
 6   PHONE                    9 non-null      object 
 7   AMOUNT_COVERED           10 non-null     float64
 8   AMOUNT_UNCOVERED         10 non-null     float64
 9   REVENUE                  10 non-null     float64
 10  COVERED_ENCOUNTERS       10 non-null     int64  
 11  UNCOVERED_ENCOUNTERS     10 non-null     int64  
 12  COVERED_MEDICATIONS      10 non-null     int64  
 13  UNCOVERED_MEDICATIONS    10 non-null     int64 

DATE                               PATIENT  \
0  2012-12-12T14:56:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
1  2015-05-04T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
2  2017-12-11T15:25:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
3  2019-01-30T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   
4  2019-05-27T11:40:19Z  3575b903-dbd0-1d55-6146-9e8aa4ed52a5   

                              ENCOUNTER       CODE  \
0  f222d3d6-d9f0-3586-70d4-69a77f16b58c   73761001   
1  c57b93a2-0302-4965-9e2c-713b84e7c24b  430193006   
2  11306017-08e3-1375-fc44-1265a8114949   73761001   
3  cd0f83e5-1e70-75ec-1be9-b254e2589e49   23426006   
4  bdeb0bc4-50bc-5aaa-ab7f-e43535a01bbe  430193006   

                                       DESCRIPTION  BASE_COST  REASONCODE  \
0                                      Colonoscopy    9209.61         NaN   
1            Medication Reconciliation (procedure)     414.03         NaN   
2                                      Colonoscopy   13437.13         NaN   
3  Measurement of respiratory function (procedure)     334.63  10509002.0   
4            Medication Reconciliation (procedure)     626.56         NaN   

             REASONDESCRIPTION  
0                          NaN  
1                          NaN  
2                          NaN  
3  Acute bronchitis (disorder)  
4                          NaN



Metadaten zu Tabelle procedures.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5815 entries, 0 to 5814
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   DATE               5815 non-null   object 
 1   PATIENT            5815 non-null   object 
 2   ENCOUNTER          5815 non-null   object 
 3   CODE               5815 non-null   int64  
 4   DESCRIPTION        5815 non-null   object 
 5   BASE_COST          5815 non-null   float64
 6   REASONCODE         3116 non-null   float64
 7   REASONDESCRIPTION  3116 non-null   object 
dtypes: float64(2), int64(1), object(5)
memory usage: 363.6+ KB


Nullwerte in Tabelle procedures.csv:
 DATE                    0
PATIENT                 0
ENCOUNTER               0
CODE                    0
DESCRIPTION             0
BASE_COST               0
REASONCODE           2699
REASONDESCRIPTION    2699
dtype: int64


Die ersten 10 Zeilen von Tabelle providers.csv:

Id                          ORGANIZATION  \
0  28e265ee-49be-3b72-b29e-3aa9a53f248e  ef58ea08-d883-3957-8300-150554edc8fb   
1  c069042b-d730-3021-8f0e-d1fa8f2bbed1  69176529-fd1f-3b3f-abce-a0a3626769eb   
2  822bc4bb-0700-3341-b343-9e5d0f7e6655  5e765f2b-e908-3888-9fc7-df2cb87beb58   
3  e7733abb-0c64-372a-b589-f2372898bfa1  f1fbcbfb-fcfa-3bd2-b7f4-df20f1b3c3a4   
4  3f3170ef-5540-31f0-9aae-e3d7384d87e3  e002090d-4e92-300e-b41e-7d1f21dee4c6   

                     NAME GENDER        SPECIALITY                  ADDRESS  \
0     Rosette746 Bruen238      F  GENERAL PRACTICE         60 HOSPITAL ROAD   
1    Janette321 Fisher429      F  GENERAL PRACTICE  330 MOUNT AUBURN STREET   
2  Chauncey770 Steuber698      M  GENERAL PRACTICE          211 PARK STREET   
3       Dagny669 Wyman904      F  GENERAL PRACTICE       ONE GENERAL STREET   
4    Caitlyn56 Medhurst46      F  GENERAL PRACTICE    1493 CAMBRIDGE STREET   

         CITY STATE    ZIP        LAT        LON  UTILIZATION  
0  LEOMINSTER    MA  01453  42.520838 -71.770876          107  
1   CAMBRIDGE    MA  02138  42.375967 -71.118275          131  
2   ATTLEBORO    MA  02703  41.931653 -71.294503            2  
3    LAWRENCE    MA  01842  42.700273 -71.161357          607  
4   CAMBRIDGE    MA  02138  42.375967 -71.118275          149



Metadaten zu Tabelle providers.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1079 entries, 0 to 1078
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Id            1079 non-null   object 
 1   ORGANIZATION  1079 non-null   object 
 2   NAME          1079 non-null   object 
 3   GENDER        1079 non-null   object 
 4   SPECIALITY    1079 non-null   object 
 5   ADDRESS       1079 non-null   object 
 6   CITY          1079 non-null   object 
 7   STATE         1079 non-null   object 
 8   ZIP           1079 non-null   object 
 9   LAT           1079 non-null   float64
 10  LON           1079 non-null   float64
 11  UTILIZATION   1079 non-null   int64  
dtypes: float64(2), int64(1), object(9)
memory usage: 101.3+ KB


Nullwerte in Tabelle providers.csv:
 Id              0
ORGANIZATION    0
NAME            0
GENDER          0
SPECIALITY      0
ADDRESS         0
CITY            0
STATE           0
ZIP  

DATE                               PATIENT  \
0  2020-02-28  28124841-1cf3-2818-d4ee-8574fac23298   
1  2020-02-28  28124841-1cf3-2818-d4ee-8574fac23298   
2  2020-02-28  28124841-1cf3-2818-d4ee-8574fac23298   
3  2020-02-28  28124841-1cf3-2818-d4ee-8574fac23298   
4  2020-02-28  28124841-1cf3-2818-d4ee-8574fac23298   

                              ENCOUNTER       CODE  \
0  ccf7943f-b369-6ab8-b9e7-dafd2526804d  409534002   
1  ccf7943f-b369-6ab8-b9e7-dafd2526804d  713779008   
2  ccf7943f-b369-6ab8-b9e7-dafd2526804d  469673003   
3  ccf7943f-b369-6ab8-b9e7-dafd2526804d  706724001   
4  ccf7943f-b369-6ab8-b9e7-dafd2526804d  419343004   

                                         DESCRIPTION  QUANTITY  
0  Disposable air-purifying respirator (physical ...         2  
1  Nitrile examination/treatment glove  non-powde...        24  
2       Isolation gown  single-use (physical object)        12  
3                      Face shield (physical object)         2  
4                   Alcohol disinfectant (substance)         1



Metadaten zu Tabelle supplies.csv:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   DATE         891 non-null    object
 1   PATIENT      891 non-null    object
 2   ENCOUNTER    891 non-null    object
 3   CODE         891 non-null    int64 
 4   DESCRIPTION  891 non-null    object
 5   QUANTITY     891 non-null    int64 
dtypes: int64(2), object(4)
memory usage: 41.9+ KB


Nullwerte in Tabelle supplies.csv:
 DATE           0
PATIENT        0
ENCOUNTER      0
CODE           0
DESCRIPTION    0
QUANTITY       0
dtype: int64


**Zusammenfassung der Auswertung**

Als möglicherweise für eine ML-Auswertung zur Vorhersage von Gesundheitskosten beeinträchtigend zeigt sich im Allgemeinen Folgendes:
*   Spalten START und STOP Datumswerte als object gespeichert sind
*   Die Spalte CODE anthaelt einen SNOMED Code und ist als float gespeichert
*   Es existieren fehelende Einträge (Null-Werte). Betroffen sind die Spalte STOP, REASONCODE,    REASONDESCRIPTION in mehreren Tabellen, sowie ENCOUNTER-ID und UNITS in der Encounter-Tabelle

Insgesamt finden sich 135 Patienten in der Patients-Tabelle. Demgegenüber stehen 7768 Einträge in der Encounter-Tabelle. Daraus laesst sich vermuten, dass entweder mehrere Fälle pro Patient existieren oder Duplikate vorliegen.

Letzteres wird wie folgt eruiert:


In [45]:
df_encounter = pd.read_csv(os.path.join(CSV_PATH, "encounters.csv"))
df_patients = pd.read_csv(os.path.join(CSV_PATH, "patients.csv"))
df_medications = pd.read_csv(os.path.join(CSV_PATH, "medications.csv"))
df_observations = pd.read_csv(os.path.join(CSV_PATH, "observations.csv"))
df_immunizations = pd.read_csv(os.path.join(CSV_PATH, "immunizations.csv"))
df_procedures = pd.read_csv(os.path.join(CSV_PATH, "procedures.csv"))


def check_for_duplicates(df, column):
  df[column] = df[column].apply(str)
  dup_df = df.duplicated(subset= [column])
  print (f"Anzahl der Duplikate in Spalte {column}:", dup_df.sum())
  print (df[dup_df].head(10))

check_for_duplicates(df_patients, "Id")
check_for_duplicates(df_medications, "ENCOUNTER")
check_for_duplicates(df_immunizations, "ENCOUNTER")
check_for_duplicates(df_observations, "ENCOUNTER")
check_for_duplicates(df_procedures, "ENCOUNTER")

Anzahl der Duplikate in Spalte Id: 0
Empty DataFrame
Columns: [Id, BIRTHDATE, DEATHDATE, SSN, DRIVERS, PASSPORT, PREFIX, FIRST, LAST, SUFFIX, MAIDEN, MARITAL, RACE, ETHNICITY, GENDER, BIRTHPLACE, ADDRESS, CITY, STATE, COUNTY, ZIP, LAT, LON, HEALTHCARE_EXPENSES, HEALTHCARE_COVERAGE]
Index: []

[0 rows x 25 columns]
Anzahl der Duplikate in Spalte ENCOUNTER: 4009
                   START                  STOP  \
5   2010-10-20T00:48:18Z  2010-12-15T00:48:18Z   
6   2010-10-20T00:48:18Z  2010-12-15T00:48:18Z   
10  2002-01-18T08:03:01Z  2003-01-24T08:03:01Z   
12  2003-01-24T08:03:01Z  2004-01-30T08:03:01Z   
14  2004-01-30T08:03:01Z  2005-02-04T08:03:01Z   
16  2005-02-04T08:03:01Z  2006-02-10T08:03:01Z   
18  2006-02-10T08:03:01Z  2007-02-16T08:03:01Z   
20  2007-02-16T08:03:01Z  2008-02-22T08:03:01Z   
22  2008-02-22T08:03:01Z  2009-02-27T08:03:01Z   
24  2009-02-27T08:03:01Z  2010-03-05T08:03:01Z   

                                 PATIENT  \
5   aff157cc-b6d3-412b-ccbe-bfd5fac1c2d5  